apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -178,7 +409,18 @@ def read_objects(
                 try:
                     local_temp_path = ""
                     parsed_path = urlparse(path, allow_fragments=False)
-                    if parsed_path.scheme != "file":
+                    if parsed_path.scheme == "tiledb":
+                        key = parsed_path.path.lstrip("/")
+                        with tiledb.open(path, "r") as a:
+                            file_extension = a.meta["file_extension"].lstrip(".")
```

Comment:
I believe you need some checks and validation that this is a TileDB File or these accesses with throw.

Reply:
Do we have a consistent way to check for file type assets? I couldn't find a type metadata tag. Should we just check for the existence of `file_extension, file_size`?

Reply:
Removed